### PR TITLE
Create endpoint to retrieve and create spaces

### LINKED ISSUE #103: (POST endpoint) For adding a space to the backend

### PR DESCRIPTION
**Pull request by jmir17**
Resolves #103 #104

### CODE REVIEW

**Comment by ifqthenp** [comment]
@takumab we decided to refactor it since `Session` will be isolated by a certain namespace/package, there should be no conflicts with classes from Java APIs

**Comment by ifqthenp** [comment]
@sherlockq Fixed

File: packages/backend/src/test/java/com/codurance/open_space/HelloWorldControllerShould.java [hunk 1]
```diff
@@ -19,10 +20,10 @@
     @Test
     void returnHelloWorld() throws Exception {
         mockMvc.perform(get("/api")
-            .contentType(MediaType.APPLICATION_JSON))
-            .andExpect(status().isOk())
-            .andExpect(content()
-                .contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
-            .andExpect(jsonPath("text", is("Hello, world!")));
+                .contentType(MediaType.APPLICATION_JSON))
```
Comment (mattgraygithub) [inline packages/backend/src/test/java/com/codurance/open_space/HelloWorldControllerShould.java]: Needs formatting

File: packages/backend/src/test/java/com/codurance/open_space/SessionControllerShould.java [hunk 2]
```diff
@@ -67,32 +70,24 @@ void post_open_space_session() throws Exception {
...
+                session.getTitle(),
+                session.getLocation(),
+                session.getTime(),
+                session.getPresenter());
+        sessionUpdated.setLocation("Location 2");
 
-        when(repository.save(openSpaceSessionUpdated))
-                .thenReturn(openSpaceSessionUpdated);
+        when(repository.save(sessionUpdated))
+                .thenReturn(sessionUpdated);
```
Comment (mattgraygithub) [inline packages/backend/src/test/java/com/codurance/open_space/SessionControllerShould.java]: Looks like this may need formatting?

**Pull request merged**